packages/xo-web/src/common/xo/index.js

```diff
@@ -1160,6 +1160,34 @@ export const convertVmToTemplate = vm =>
     ),
   }).then(() => _call('vm.convert', { id: resolveId(vm) }), noop)
 
+export const cloneToTemplate = async vm => {
+  const vmType = vm.type === 'VM-snapshot' ? 'snapshot' : 'VM'
+  await confirm({
+    title: _('cloneAsTemplate'),
+    body: (
+      <div>
+        <p>{_('cloneAsTemplateMessage', { type: vmType })}</p>
+      </div>
+    ),
+  })
+  try {
+    await _call('vm.cloneToTemplate', { id: resolveId(vm) }, noop)
+  } catch (reason) {
+    if (
+      (incorrectState.is(reason),
+      {
+        data: {
+          expected: 'Halted',
+          property: 'power_state',
+        },
+      })
+    ) {
+      await stopVm(vm)
```

Comment:
I think we should ask the user to confirm before shutting down the VM. And this should probably be implemented in the server so that the VM can be automatically restarted afterwards.

Reply:
[stopVm](https://github.com/vatesfr/xen-orchestra/blob/75e3e36aa87f4f553c9307dec5b36ddd3075c03f/packages/xo-web/src/common/xo/index.js#L1080) ask confirmation to the user before stop the VM.